lib/shopping_cart.rb

```diff
@@ -0,0 +1,29 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity= capacity.to_i
+    @products = []
+end
+
+def add_products(product)
+
+  @products << product
+end
+
+def total_number_of_products
+  @quantity.count
+end
+
+def details
+  hash_cart = {name: @name, capacity: @capacity}
+end
+
+def total_number_of_products
+  if total_number_of_products > 30
+    @full = true
+  end
+end
```

Comment:
Careful, you're naming this method the same as a previous method. You're hard coding that 30 on line 24, where I think you may be referring to the capacity. Make sure to use variables instead of the actual hard coded value to make your code more dynamic.